rllib/utils/minibatch_utils.py

```diff
@@ -59,17 +59,20 @@ def __iter__(self):
             minibatch = {}
             for module_id, module_batch in self._batch.policy_batches.items():
                 s = self._start[module_id]  # start
-                e = s + self._minibatch_size  # end
+                n_steps = self._minibatch_size
 
                 samples_to_concat = []
                 # cycle through the batch until we have enough samples
-                while e >= len(module_batch):
-                    samples_to_concat.append(module_batch[s:])
-                    e = self._minibatch_size - len(module_batch[s:])
+                while n_steps >= len(module_batch[s:]):
```

Comment:
how does this handle the base case of module_batch having 0 timesteps

Reply:
module_batch[s:] will create a new list for each iteration.
small performance penalty that may be unnecessary.
also, we can stay with the s and e model I feel like, just need to flip the >= to < right? :)

Reply:
if module_batch is empty, then we will never exit this while loop, unless n_steps is 0, or negative, AFAICT

Reply:
> how does this handle the base case of module_batch having 0 timesteps
module_batch should never be empty. Logically that should actually create an infinite loop. I will add a check for this.

Reply:
> module_batch[s:] will create a new list for each iteration.

Fixed it via `while n_steps >= len(module_batch) - s`

Reply:
> also, we can stay with the s and e model I feel like, just need to flip the >= to < right? :)

This way of writing it was more intuitive to me at least. After I figured out the cause for infinit loop.

Reply:
it's done. We do now.

Reply:
yay